common/components/BalanceSidebar/index.tsx

```diff
@@ -77,9 +89,16 @@ export class BalanceSidebar extends React.Component<Props, {}> {
         name: 'Token Balances',
         content: (
           <TokenBalances
-            tokens={tokenBalances}
+            tokens={tokens}
```

Comment:
Consider connecting the component directly, this is kind of props overload here. And for `walletTokens` the boolean logic can be deferred to a selector